src/extension/prompts/node/agent/xAIPrompts.tsx

```diff
@@ -121,6 +123,10 @@ class DefaultGrokCodeFastAgentPrompt extends PromptElement<DefaultAgentPromptPro
 class XAIPromptResolver implements IAgentPrompt {
 	static readonly familyPrefixes = ['grok-code'];
 
+	static async matchesModel(endpoint: IChatEndpoint): Promise<boolean> {
+		return endpoint.family === BYOK_PROVIDER_NAME;
```

Comment:
What is this used for? It seems unrelated to the rest of the PR

Reply:
I can move this to a separate PR, this just enables the xAI prompts for the BYOK models as well. I tested that the behaviour for models like Grok 4 and other BYOK are relatively the same after the change and nothing is broken.